### DISCUSSION TITLE
Wrong topic send via tele message?  Sends old value or initial default value not new static value?

### DISCUSSION THREAD

**Discussion by dachshund-digital** [arendst/Tasmota#12128] [Q&A] [answered]
Has anyone else noticed that the 'tele' resulting message has the wrong topic?  I changed the topic to an explicit string 'Test' in the MQTT configuration of a given tasmota flashed device (firmware 9.4.0).  I then let the tele message result.  I happen to be watching this via termite and via Node Red flow.  The tele messages are sending the old topic value, it almost seems like the Tasmota code is pulling the topic from somewhere else, not sure where, because no where in the configuration view or information views can I see the topic as anything for MQTT then 'Test' as set statically.

I have re-flashed, I have restarted the device, I have changed the topic to Test, Test1, Test2, etc.  And still the old default topic is returned as it was once set by the initial boot after flashing.  Anyone have any ideas why this is happening?  I have seen the behavior on older 8.3.1 firmware as well.

**Comment by sfromis** [comment, marked as answer]
To test, I changed topic name on a device with Tasmota. As expected, this resulted in restarting with the new topic in place in telemetry messages, subscriptions and other payloads. Even turning off savedata, changing topic still worked as it should. 

One "surprise" could be if you try have same Topic and MqttClient. As this is [not allowed](https://tasmota.github.io/docs/Commands/#topic), your topic change will be reverted.

**Comment by dachshund-digital** [comment]
So the client id for MQTT can't be the topic string?  This works just fine in other MQTT implementations, python3 for example.

**Reply by barbudor** [reply]
Is this your case ? 
You don't give any details such as configuration, status 0, logs, etc ... 
Copying here your Termite logs when you do such change with `SerialLog 3` enabled may help
So we can just guess your use case and prove that in our case it works just fine
You may have configured Tasmota for not saving the configuration (savedata)

For the ClientID question, there might be historical reasons (currently investigating). One point is that Tasmota automatically create a fallback topic based on the client ID (`cmnd/<ClientID>_fb`) so having both main topic and fallback topic similar seems to loose the interest of a fallback topic based on the mac address (which is default ClientID).

**Reply by dachshund-digital** [reply]
Nope not fb topic.  I left that unchanged as a test point.  As for the 'tele' messages being the wrong topic, there was no doubt at the time I opened the issue I was seeing the original default topic still used after I customized it.

Moreover, as of this morning, the tele messages are now showing the correct topic.  The only change I did yesterday was change the topic to something unique, but different from before, ensure I had the configuration backed up and then reboot ESP.  I then ignored the ESP for 24 hours given the holiday yesterday.  This morning watching the messages, now the latest topic is being used.

Odd, really odd.

**Comment by dachshund-digital** [comment]
Was unable to recreate the issue consistently.  Closed the issue please.
